controllers/nim_account_controller.go

```diff
@@ -0,0 +1,480 @@
+		return ctrl.Result{}, err
+	} else {
+		ref, refErr := reference.GetReference(r.Scheme(), cmap)
+		if refErr != nil {
+			return ctrl.Result{}, refErr
+		}
+		targetStatus.NIMConfig = ref
+	}
+	dataCmapOk := "data config map reconciled successfully"
+	logger.V(1).Info(dataCmapOk)
+	meta.SetStatusCondition(&targetStatus.Conditions, makeAccountFailureCondition(account.Generation, dataCmapOk))
+	meta.SetStatusCondition(&targetStatus.Conditions, makeConfigMapSuccessfulCondition(account.Generation, dataCmapOk))
+
+	// reconcile template
+	if template, err := r.reconcileRuntimeTemplate(ctx, account); err != nil {
+		msg := "runtime template reconcile failed"
+		logger.V(1).Error(err, msg)
+		meta.SetStatusCondition(&targetStatus.Conditions, makeAccountFailureCondition(account.Generation, msg))
+		meta.SetStatusCondition(&targetStatus.Conditions, makeTemplateFailureCondition(account.Generation, msg))
+		return ctrl.Result{}, err
+	} else {
+		ref, refErr := reference.GetReference(r.Scheme(), template)
+		if refErr != nil {
+			return ctrl.Result{}, refErr
+		}
+		targetStatus.RuntimeTemplate = ref
+	}
+	templateOk := "runtime template reconciled successfully"
+	logger.V(1).Info(templateOk)
+	meta.SetStatusCondition(&targetStatus.Conditions, makeAccountFailureCondition(account.Generation, templateOk))
+	meta.SetStatusCondition(&targetStatus.Conditions, makeTemplateSuccessfulCondition(account.Generation, templateOk))
+
+	// reconcile pull secret
+	if pullSecret, err := r.reconcileNimPullSecret(ctx, ownerRefCfg, account.Namespace, apiKeyStr); err != nil {
+		msg := "pull secret reconcile failed"
+		logger.V(1).Error(err, msg)
+		meta.SetStatusCondition(&targetStatus.Conditions, makeAccountFailureCondition(account.Generation, msg))
+		meta.SetStatusCondition(&targetStatus.Conditions, makePullSecretFailureCondition(account.Generation, msg))
+		return ctrl.Result{}, err
+	} else {
+		ref, refErr := reference.GetReference(r.Scheme(), pullSecret)
+		if refErr != nil {
+			return ctrl.Result{}, refErr
+		}
+		targetStatus.NIMPullSecret = ref
+	}
+	pullSecOk := "pull secret reconciled successfully"
+	logger.V(1).Info(pullSecOk)
+	meta.SetStatusCondition(&targetStatus.Conditions, makeAccountSuccessfulCondition(account.Generation, "reconciled successfully"))
+	meta.SetStatusCondition(&targetStatus.Conditions, makePullSecretSuccessfulCondition(account.Generation, pullSecOk))
+
+	return ctrl.Result{}, nil
+}
+
+// reconcileNimConfig is used for reconciling the configmap encapsulating the model data used for constructing inference services
+func (r *NimAccountReconciler) reconcileNimConfig(
+	ctx context.Context, ownerCfg *ssametav1.OwnerReferenceApplyConfiguration,
+	namespace, apiKey string, runtimes []utils.NimRuntime,
+) (*corev1.ConfigMap, error) {
+	data, dErr := utils.GetNimModelData(apiKey, runtimes)
+	if dErr != nil {
+		return nil, dErr
+	}
+
+	cmapCfg := ssacorev1.ConfigMap(constants.NimDataConfigMapName, namespace).
```

Comment:
nit: cmap -> cm

Reply:
Resolved in https://github.com/opendatahub-io/odh-model-controller/pull/289/commits/b8293f4d86d0821c78a8b52c56a7720f8ed4228f.